Security considerations: the identity element (#306)
cfrg/draft-irtf-cfrg-hash-to-curve: draft-irtf-cfrg-hash-to-curve.md

When the required encoding is not clear, applications SHOULD use a uniform
encoding.

Both encodings given in {{roadmap}} can output the identity element of the group G.

> Assuming that the encoding is instantiated with a hash\_to\_field function
> that follows all guidelines in {{hashtofield}}, for a uniformly random input
> to either encoding the probability that the identity element is output is
> roughly 1/r, which is negligible for cryptographically useful elliptic curves.

Reviewer: ```suggestion
The probability that either encoding function, when instantiated with a hash\_to\_field function 
that follows all guidelines in {{hashtofield}}, outputs the identity element given a uniformly
random input is roughly 1/r, which is negligible for cryptographically useful elliptic curves.
```

Author: Your suggestion definitely helps, but I wonder if we can make this even clearer. How about:

```
The probability that either encoding function outputs the identity element on a random input is roughly 1/r, which is negligible for cryptographically useful elliptic curves.
Further, it is computationally infeasible to find an input to either encoding function whose corresponding output is the identity element.
(Both of these properties hold when the encoding functions are instantiated with a hash\_to\_field function that follows all guidelines in {{hashtofield}}.)
```

Maybe something like this is clearer?

Reviewer: Much better! :-)

Author: OK, pushed that change now. Marking this resolved (but happy to discuss more!)

> Further, it is computationally infeasible to find an input to either encoding

Reviewer: ```suggestion
Further, it is computationally infeasible to find an input to either encoding function
```

whose corresponding output is the identity element.

> In general, protocols that use these encoding functions SHOULD NOT add a
> special case to detect and "fix" the identity element, because implementation
> errors in this special-case logic are astronomically more likely than a
> "bad" input ever occurring.

Reviewer: ```suggestion
Protocols that use these encoding functions SHOULD NOT add a special case 
to detect and "fix" the identity element.
```
